var/spack/repos/builtin/packages/hdf-eos5/package.py

```diff
@@ -92,7 +97,10 @@ def configure_args(self):
 
         # Provide config args for dependencies
         extra_args.append('--with-hdf5={0}'.format(self.spec['hdf5'].prefix))
-        if self.spec['zlib']:
+        if 'libszip' in self.spec:
```

Comment:
I've recently learned that this (and below) should probably be the virtual  package `szip`, of which `libszip` is the default implementation but `libaec` is an alternative. See the discussion in https://github.com/spack/spack/pull/18937 .